src/initializers/kafka/base-kafka-handler.ts

```diff
@@ -0,0 +1,43 @@
+import { KafkaMessage, NConsumer } from 'sinek';
+import Kafka from './kafka';
+import { Logger } from 'log4js';
+import { flatten } from 'lodash';
+
+export default abstract class BaseKafkaHandler {
+  consumer: NConsumer;
+  topic: string;
+  batchSize: number;
+  logger: Logger;
+
+  constructor(kafka: Kafka, options: { topic: string; logger: Logger; batchSize?: number; }) {
+    const { topic, batchSize, logger } = options;
+    this.topic = topic;
+    this.batchSize = batchSize || 1;
+    this.logger = logger;
+    this.consumer = kafka.createConsumer(this.topic);
+    this.consumer.connect().then(() => this.consume().catch(err => this.logger.error(err)));
```

Comment:
If this fails the instance is invalid. Can you move object instantiation to an async factory method instead?

Reply:
I can remove the `consumer.connect()` from the constructor and add it in the `consume`. Then the caller must call `consume()` explicitly. Does that work?

Reply:
What if the connection fails after a while? Can we handle reconnections somehow?

Reply:
Supposedly it handles reconnections automatically
```
      this.consumer.on("disconnected", () => {
        if(this._inClosing){
          this._reset();
        }
        logger.warn("Disconnected.");
        //auto-reconnect --> handled by consumer.consume();
      });
```

I'll have to test locally (terminate the broker and bring it back up or something equivalent)

Reply:
Used this https://hub.docker.com/r/landoop/kafka-lenses-dev/ locally in order to test what happens when terminating the broker. 
Everything works as expected. Both the consuming and producing resume automatically.